drivers/adc/adc_demo/adc_demo.h

```diff
@@ -50,22 +50,34 @@
 /*************************** Types Declarations *******************************/
 /******************************************************************************/
 
-#define MAX_REG_ADDR		16
+#define MAX_ADC_ADDR		16
+#define DEFAULT_LOCAL_SAMPLES 128
+#define SINE_VECTOR_SIZE 128
 
 /**
  * @struct iio_demo_adc_desc
  * @brief Desciptor.
  */
 struct adc_demo_desc {
 	/** Dummy registers of device for testing */
-	uint8_t reg[MAX_REG_ADDR];
+	uint8_t reg[MAX_ADC_ADDR];
+	/** Active channel**/
+	uint32_t active_ch;
+	/** buffer for communication between adc&dac, according to channel*/
+	uint16_t **loopback;
+//	/* NUmber of channels*/
+//	uint32_t channel_no;
 };
 
 /**
  * @struct iio_demo_adc_init_param
  * @brief iio demo adc configuration.
  */
 struct adc_demo_init_param {
+	/* NUmber of channels*/
+	uint32_t channel_no;
+	/**buffer for communication between adc&dac, according to appropriate channel*/
```

Comment:
I think this can be described as array of buffers for each channel.

Reply:
Did you mean in the comment description?

Reply:
yes